**Step-1:Import packages**

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


**step-2:Read csv**

In [2]:
visa_df=pd.read_csv(r"C:\Users\B VENKATA SAILAJA\Documents\Datasets_GenAi\Visadataset.csv")
visa_df

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


**separate cata and nume Columns**

In [6]:
cat=visa_df.select_dtypes(include='object').columns.tolist()
cat

['case_id',
 'continent',
 'education_of_employee',
 'has_job_experience',
 'requires_job_training',
 'region_of_employment',
 'unit_of_wage',
 'full_time_position',
 'case_status']

In [7]:
num=visa_df.select_dtypes(exclude='object').columns.tolist()
num

['no_of_employees', 'yr_of_estab', 'prevailing_wage']

- frequency table
- bar chat
- relative frequency table
- pie chart

**Step-3:Frequency table**

In [8]:
cat

['case_id',
 'continent',
 'education_of_employee',
 'has_job_experience',
 'requires_job_training',
 'region_of_employment',
 'unit_of_wage',
 'full_time_position',
 'case_status']

In [10]:
visa_df['continent']

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

In [13]:
visa_df['continent'].unique()
#to know the repeting values

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [15]:
visa_df['continent'].nunique()#how many times repeted

6

- There are 25480 applicants from 6 countries
- number of applicants from each continent: frequency table

In [20]:
#number of applicants from 'Asia'
visa_df#df
visa_df['continent']#one column
con=visa_df['continent']=='Asia'#one class
np.sum(con)#1+1+1+1+0+1.......... tells how many times it repeted

np.int64(16861)

In [19]:
visa_df[con]#to extract the only asia rows

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
5,EZYV06,Asia,Master's,Y,N,2339,2012,South,78252.1400,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


In [22]:
con1=visa_df['continent']=='Africa'
np.sum(con1)

np.int64(551)

In [23]:
visa_df[con1]

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.39,Year,Y,Certified
18,EZYV19,Africa,Master's,Y,N,4743,2004,Midwest,150441.13,Year,Y,Certified
74,EZYV75,Africa,Master's,Y,N,3705,2001,South,47170.76,Year,Y,Certified
194,EZYV195,Africa,Master's,Y,N,2180,1992,Midwest,113637.40,Year,Y,Certified
242,EZYV243,Africa,Bachelor's,N,Y,2509,2003,West,51886.04,Year,Y,Denied
...,...,...,...,...,...,...,...,...,...,...,...,...
25385,EZYV25386,Africa,Doctorate,Y,N,2513,1975,West,76348.20,Year,Y,Certified
25408,EZYV25409,Africa,Master's,Y,Y,1671,1992,Midwest,55756.35,Year,Y,Certified
25443,EZYV25444,Africa,Bachelor's,N,N,72892,2007,Northeast,215.10,Hour,Y,Certified
25446,EZYV25447,Africa,Master's,N,Y,2024,1971,South,42353.45,Year,Y,Certified


In [24]:
con2=visa_df['continent']=='North America'
np.sum(con2)

np.int64(3292)

In [25]:
visa_df[con2]

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
7,EZYV08,North America,Bachelor's,Y,N,3035,1924,West,418.2298,Hour,Y,Denied
21,EZYV22,North America,Master's,Y,N,1706,2013,Midwest,137694.9800,Year,N,Certified
23,EZYV24,North America,High School,N,N,1517,1884,Midwest,90108.3800,Year,N,Certified
25,EZYV26,North America,Master's,N,N,1100,1997,Midwest,99489.6500,Year,Y,Certified
43,EZYV44,North America,Master's,Y,N,2582,1989,South,54614.4200,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25451,EZYV25452,North America,High School,Y,N,97,2013,Northeast,111518.5900,Year,N,Denied
25465,EZYV25466,North America,High School,N,N,2577,1995,South,481.2238,Hour,Y,Certified
25467,EZYV25468,North America,Master's,N,Y,2504,1882,Northeast,105900.6600,Year,Y,Certified
25469,EZYV25470,North America,Master's,Y,N,2261,1997,Northeast,273772.4700,Year,N,Certified


In [35]:
cont_class=visa_df['continent'].unique()
freq=[]
for i in cont_class:
    con=visa_df['continent']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),columns=['class','Frequency'])

,class,Frequency
0,Asia,16861
1,Africa,551
2,North America,3292
3,Europe,3732
4,South America,852
5,Oceania,192


**Value_Counts**

In [36]:
visa_df['continent'].value_counts()

continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64

In [37]:
type(visa_df['continent'].value_counts())

pandas.core.series.Series

In [40]:
keys=visa_df['continent'].value_counts().keys().tolist()
keys

['Asia', 'Europe', 'North America', 'South America', 'Africa', 'Oceania']

In [41]:
values=visa_df['continent'].value_counts().values.tolist()
values

[16861, 3732, 3292, 852, 551, 192]

In [42]:
pd.DataFrame(zip(keys,values),columns=['Class','Frequency'])

,Class,Frequency
0,Asia,16861
1,Europe,3732
2,North America,3292
3,South America,852
4,Africa,551
5,Oceania,192


In [44]:
list(visa_df.groupby(by='continent'))

[('Africa',
           case_id continent education_of_employee has_job_experience  \
  4         EZYV05    Africa              Master's                  Y   
  18        EZYV19    Africa              Master's                  Y   
  74        EZYV75    Africa              Master's                  Y   
  194      EZYV195    Africa              Master's                  Y   
  242      EZYV243    Africa            Bachelor's                  N   
  ...          ...       ...                   ...                ...   
  25385  EZYV25386    Africa             Doctorate                  Y   
  25408  EZYV25409    Africa              Master's                  Y   
  25443  EZYV25444    Africa            Bachelor's                  N   
  25446  EZYV25447    Africa              Master's                  N   
  25474  EZYV25475    Africa             Doctorate                  N   
  
        requires_job_training  no_of_employees  yr_of_estab  \
  4                         N             1082

In [47]:
visa_df.groupby(by='continent').count()

,case_id,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
continent,,,,,,,,,,,
Africa,551,551,551,551,551,551,551,551,551,551,551
Asia,16861,16861,16861,16861,16861,16861,16861,16861,16861,16861,16861
Europe,3732,3732,3732,3732,3732,3732,3732,3732,3732,3732,3732
North America,3292,3292,3292,3292,3292,3292,3292,3292,3292,3292,3292
Oceania,192,192,192,192,192,192,192,192,192,192,192
South America,852,852,852,852,852,852,852,852,852,852,852


In [48]:
visa_df.groupby(by='continent').count()['case_id']

continent
Africa             551
Asia             16861
Europe            3732
North America     3292
Oceania            192
South America      852
Name: case_id, dtype: int64